src/odemis/model/_dataflow.py

```diff
@@ -105,7 +105,7 @@ class DataFlowBase(object):
     """
     def __init__(self):
         self._listeners = set()
-        self._lock = threading.Lock() # need to be acquired to modify the set
+        self._lock = threading.RLock()  # need to be acquired to modify the set
```

Comment:
In general, when modifying the model, it's better to do it via a separate commit, with a large explanation.